ricecooker/utils/youtube_cache.py

```diff
@@ -0,0 +1,167 @@
+import json
+import os
+import re
+import youtube_dl
+
+from pressurecooker.youtube import YouTubeResource
+from ricecooker.config import LOGGER
+
+
+# CONSTANTS for YouTube cache
+################################################################################
+CHEFDATA_DIR = 'chefdata'
+DEFAULT_YOUTUBE_CACHE_DIR = os.path.join(CHEFDATA_DIR, 'youtubecache')
+
+
+# CONSTANTS for YouTube resources
+################################################################################
+YOUTUBE_VIDEO_REGEX = re.compile(
+    r'(https?://)?(www\.)?(youtube|youtu|youtube-nocookie)\.(com|be)/(watch\?v=|embed/|v/|.+\?v=)?(?P<youtube_id>[A-Za-z0-9\-=_]{11})'
+)
+YOUTUBE_PLAYLIST_URL_FORMAT = "https://www.youtube.com/playlist?list={0}"
+YOUTUBE_VIDEO_URL_FORMAT = "https://www.youtube.com/watch?v={0}"
+
+
+class YouTubeVideoCache(object):
+
+    def __init__(self, video_id, alias='', cache_dir=''):
+        """
+        Initializes YouTubeVideoCache object with video_id
+        :param alias: Alias name for the JSON cache filename, which will be named as youtube_id if such field not specified
+        """
+        self.video_id = video_id
+        self.url = YOUTUBE_VIDEO_URL_FORMAT.format(self.video_id)
+        if not alias:
+            self.cachename = self.video_id
+        else:
+            self.cachename = alias
+        if not cache_dir:
+            self.cache_dir = DEFAULT_YOUTUBE_CACHE_DIR
+        else:
+            self.cache_dir = cache_dir
+        if not os.path.isdir(self.cache_dir):
+            os.mkdir(self.cache_dir)
```

Comment:
I would recommend using `os.makedirs(self.cache_dir, exist_ok=True)` instead. This way, you don't have to do the `isdir` check, and it will also create any parent directories of `self.cache_dir` that are missing as well.